Add style guide (#624)
software-mansion/starknet.py: docs/development.rst

    # Run only e2e tests
    poe test_e2e

Running e2e tests in PyCharm
----------------------------
1. Run ``starkware-devnet`` script before running e2e tests in PyCharm.
2. Use ``E2E tests`` configuration to run or debug.

⚠️ **Warning**: Make sure to fill your interpreter in the configuration, to match your project's poetry venv.

Code style guide
----------------

Rules to follow when writing a code:

1. Check the code with pylint

.. code-block:: bash

    poe lint

2. Format the code with black

.. code-block:: bash

    poe format

3. Run a typechecker (pyright)

.. code-block:: bash

    poe typecheck

4. Add constant values to the constants.py file.

> 5. Add keyword-only arguments when necessary.

Reviewer: ```suggestion
5. Prefer keyword-only arguments where appropriate.
```

> 6. All classes should be marked with the `@add_sync_methods` decorator.

Reviewer: ```suggestion
6. All public classes providing async api should be marked with the `@add_sync_methods` decorator.
```

> 7. Error messages should start with a capital letter.

Reviewer: Let's also write something about writing `Argument x is ...` instead of `X is ...`.

> 8. All sentences (in docstrings/errors/comments?) should be ended with a dot.

Reviewer: ```suggestion
8. All sentences (in docstrings/errors/comments?) should be ended with a period.
```
or `full stop`.

Reviewer: I'm not sure if `(in docstrings/errors/comments?)` is necessary.

Author: Not sure too. Should sentences in the comments be ended with a period? It's not always necessary

> 9. TODO: rules about the comments

Reviewer: What about them?

Author: We should think about it. I don't remember if we talked about comments style. Any ideas? @cptartur @drknzz

> 8. Use `Argument x is...` instead of `X is...` when constructing error messages.

Reviewer: ```suggestion
8. Use `Argument x is...` instead of `X is...` when error message starts with argument (property) name.
```
How about this?
